src/reanimated2/animations.js

```diff
@@ -63,12 +63,25 @@ export function withTiming(toValue, userConfig, callback) {
     }
 
     function start(animation, value, now, previousAnimation) {
-      animation.startTime = now;
-      animation.progress = 0;
+      if (
+        previousAnimation &&
+        previousAnimation.type === 'timing' &&
+        previousAnimation.toValue === toValue
```

Comment:
Would it make sense to check also duration and easing?

Reply:
It isn't as easy because easing is a function and will be instantiated each time when you start, so we cannot compare it. Regardless, it is debatable whether changing easing should restart the animation. When new easing is assigned, the animation would adapt the new easing curve, just won't start from the beggining. I think it is fine because when you want to restart animation there is a workaround and you can use `cancelAnimation`, there is no way to prevent restart if you don't want it so I'd rather err on the side of not restarting in some cases.